server/files/configure_misp.sh

```diff
@@ -101,26 +103,27 @@ apply_updates() {
 init_user() {
     # Create the main user if it is not there already
     sudo -u www-data /var/www/MISP/app/Console/cake userInit -q 2>&1 > /dev/null
-    echo "... setting admin email to '${ADMIN_EMAIL}'"
-    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "MISP.email" ${ADMIN_EMAIL}
+
     echo "UPDATE misp.users SET email = \"${ADMIN_EMAIL}\" WHERE id = 1;" | ${MYSQLCMD}
+
     if [ ! -z "$ADMIN_ORG" ]; then
         echo "UPDATE misp.organisations SET name = \"${ADMIN_ORG}\" where id = 1;" | ${MYSQLCMD}
     fi
 
-    if [ "$AUTOCONF_ADMIN_KEY" == "true" ]; then
-        if [ ! -z "$ADMIN_KEY" ]; then
-            echo "... setting admin key to '${ADMIN_KEY}'"
-            CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1 "${ADMIN_KEY}")
-        else
-            echo "... regenerating admin key (set \$ADMIN_KEY if you want it to change)"
-            CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1)
-        fi
-        ADMIN_KEY=`${CHANGE_CMD[@]} | awk 'END {print $NF; exit}'`
-        echo "... admin user key set to '${ADMIN_KEY}'"
+    if [ -n "$ADMIN_KEY" ]; then
+        echo "... setting admin key to '${ADMIN_KEY}'"
+        CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1 "${ADMIN_KEY}")
+    elif [ -z "$ADMIN_KEY" ] && [ "$AUTOGEN_ADMIN_KEY" == "true" ]; then
+        echo "... regenerating admin key (set \$ADMIN_KEY if you want it to change)"
+        CHANGE_CMD=(sudo -u www-data /var/www/MISP/app/Console/cake User change_authkey 1)
     else
         ADMIN_KEY=""
-        echo "... admin user key auto configuration disabled"
+        echo "... admin user key auto generation disabled"
+    fi
+
+    if [[ -v CHANGE_CMD[@] ]]; then
```

Comment:
Why -v CHANGE_CMD[@] and not -v CHANGE_CMD?

Reply:
My best response is "that is how shellcheck said it should be done"

Reply:
When referencing arrays, $myarray is equivalent to ${myarray[0]} -- which is usually the first of multiple elements. This is also true for associative arrays. Therefore, if 0 (zero) is not a valid key, $myarray expands to an empty string.

To get all elements as separate parameters, use the index @ (and make sure to double quote). In the example, echo "${myarray[@]}" is equivalent to echo "foo" "bar".

Reply:
Above is from shellcheck https://www.shellcheck.net/wiki/SC2128 - basicly it makes sure not only the first, possibly empty element is returned. Should I had that explanation and a link to the code as comment?

Reply:
@Kagee so in this context we use @ so that we can test that each string inside CHANGE_CMD is a valid string (non empty)?

Reply:
-v checks if the  variable is seto to anything .... the @ is for shellcheck to shut up ... In theory, if i understand correctly, it is the difference of checking if the first element is set, or if "the array" is set.